src/main/java/org/dataloader/stats/StatisticsCollector.java

```diff
@@ -0,0 +1,49 @@
+package org.dataloader.stats;
+
+/**
+ * This allows statistics to be collected for {@link org.dataloader.DataLoader} operations
+ */
+public interface StatisticsCollector {
```

Comment:
not sure if it is really needed to always return then current value after the "action": the class itself itself is designed to be stateful and you can always call `getStatistics`.